compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -475,6 +476,28 @@ convertSimpleRecordDef env tycon = do
 
     pure $ typeDef : [workerDef | flavour == NewtypeFlavour]
 
+convertTypeSynonym :: Env -> TyCon -> ConvertM [Definition]
+convertTypeSynonym env tycon
+    | NameIn DA_Generics _ <- GHC.tyConName tycon
+    = pure []
+
+    | envLfVersion env `supports` featureTypeSynonyms
+    , Just (params, body) <- synTyConDefn_maybe tycon
+    , isLiftedTypeKind (tyConResKind tycon)
+    , not (isKindTyCon tycon)
+    = do
+        (env', tsynParams) <- bindTypeVars env params
+        tsynType <- convertType env' body
+        let tsynName = mkTypeSyn [getOccText tycon]
+        case tsynType of
+            TUnit -> pure []
+                -- ^ We avoid converting TUnit type synonyms because it
+                -- clashes with the conversion of empty typeclasses.
```

Comment:
Aren't we translating empty type classes into empty structural records? Isn't that different from the unit type?

Reply:
LF does not allow for empty structural records.

Reply:
I wonder why we decided to do it like that. I have no recollection... 🤔